### ISSUE TITLE
Using openvdb for CSG and level of detail baking

### ISSUE THREAD

**Issue by fire** [godotengine/godot#18110]
**Godot version:**
master

**OS/device including version:**
any

**Issue description:**
Feature request:

http://www.openvdb.org/documentation/doxygen/codeExamples.html#sCsgTools

https://github.com/fxtentacle/openvdb-csg

Reduz mentioned this approach is probably not the preferred core engine method. Reasoning because of possible interactivity problems and memory usage. This has not been tested though.

Openvdb can be used for both mesh CSG operations (mesh -> levelset (voxels) -> mesh).

Level of detail transforms can also be done with the same operation except setting the adaptivity to be less than 1. PBR material attributes can be converted to Surface Attributes.

It should be possible to be used in gdnative.

In summary, openvdb allows both CSG and level of detail mesh generation with UV remapping.

**Comment by fire** [comment]
CSG is implemented. Level of detail is done both https://github.com/godot-extended-libraries/mesh_lod and https://github.com/HungryProton/concept_graph/issues/44.

Closing.
